pkg/beacon/relay/chain/group_ticket.go

```diff
@@ -1,13 +1,22 @@
 package chain
 
-import "math/big"
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+)
 
 // Ticket represents group selection ticket as seen on-chain.
 type Ticket struct {
 	Value *big.Int // W_k
 	Proof *TicketProof
 }
 
+// ValueBytes returns ticket value as a byte slice.
+func (t *Ticket) ValueBytes() []byte {
+	return common.LeftPadBytes(t.Value.Bytes(), 8)[:8]
+}
```

Comment:
Instead of this function, why don't we just replace `Value *big.Int` with `Value [8]byte`? 

The conversion to and from `*big.Int` scares me as we may run into problems with byte ordering.

Reply:
Yeah, you are right. Good point!

Reply:
Can we remove this function? If `local.go` uses this conversion, let's have it in `local.go` where there is no risk with endianness problems.

Reply:
This function is used only in `local.go`. I think it makes sense to do the conversion there and remove it from here. Especially that bytes->uint256 conversion is done on-chain. This will work as an additional safety guard for byte endianness. Let's be specific about it in `local.go`.

```
sort.SliceStable(c.tickets, func(i, j int) bool {
  // ticket value bytes are interpreted as a big-endian unsigned integers
  iValue := new(big.Int).SetBytes(c.tickets[i].Value[:])
  jValue := new(big.Int).SetBytes(c.tickets[j].Value[:])

  return iValue.Cmp(jValue) == -1
})
```

and later:
```
_ = promise.Fulfill(&event.GroupTicketSubmission{
  TicketValue: new(big.Int).SetBytes(ticket.Value[:]),
  BlockNumber: c.simulatedHeight,
})
```

And the same in `groupselection_test.go`:
```
	sort.SliceStable(stg.submittedTickets, func(i, j int) bool {
		// ticket value bytes are interpreted as a big-endian unsigned integers
		iValue := new(big.Int).SetBytes(stg.submittedTickets[i].Value[:])
		jValue := new(big.Int).SetBytes(stg.submittedTickets[j].Value[:])

		return iValue.Cmp(jValue) == -1
	})

	// (...)

	_ = promise.Fulfill(&event.GroupTicketSubmission{
		TicketValue: new(big.Int).SetBytes(ticket.Value[:]),
		BlockNumber: 222,
	})
```

and finally `ticket_submission_test.go`:
```
TicketValue: new(big.Int).SetBytes(t.Value[:]),
```

My point here is that we shouldn't have this function here, it's used only for tests. And tests should ensure the input matches the specification (big-endian, unsigned integer)

Reply:
I added this method just for convenience but you're right. I will replace as you suggested.